File: Python/232. Implement Queue using Stacks.py
class MyQueue:

    def __init__(self):
        self.inSt = []
        self.outSt = []
    def push(self, x: int) -> None:
        self.inSt.append(x)

    def pop(self) -> int:
        if self.outSt:
            return self.outSt.pop(-1)
        while self.inSt:
            self.outSt.append(self.inSt.pop(-1))
        return self.outSt.pop(-1)

    def peek(self) -> int:
        res = self.pop()
        self.outSt.append(res)
        return res

    def empty(self) -> bool:
        return not self.inSt and not self.outSt


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: C++/232. Implement Queue using Stacks.cpp
class MyQueue {
public:
    stack<int>inputStack;
    stack<int>outputStack;
    MyQueue() {
    }

    void push(int x) {
        inputStack.push(x);
    }

    int pop() {
        if (!outputStack.empty()) {
            int res = outputStack.top();
            outputStack.pop();
            return res;
        }
        else {
            while (!inputStack.empty()) {
                int temp = inputStack.top(); inputStack.pop();
                outputStack.push(temp);
            }
            if (!outputStack.empty()) {
                int res = outputStack.top();
                outputStack.pop();
                return res;
            }
            else return false;
        }
    }

    int peek() {
        int res = this->pop();
        outputStack.push(res);
        return res;
    }

    bool empty() {
        return inputStack.empty() && outputStack.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */